engine/function/src/templates/Numeric.ftl

```diff
@@ -0,0 +1,2358 @@
+            if (!isNull(c) && !isNull(w)) {
+                vsum += c * w;
+                wsum += w;
+            }
+        }
+        return vsum / wsum;
+    }
+
+    </#if>
+    </#list>
+
+
+    /**
+     * Returns a sequence of values.
+     *
+     * @param start starting value.
+     * @param end terminal value.
+     * @param step step size.
+     * @return sequence of values from start to end.
+     */
+    public static ${pt.primitive}[] sequence(${pt.primitive} start, ${pt.primitive} end, ${pt.primitive} step) {
+        if(step == 0) {
+            return new ${pt.primitive}[0];
+        }
+
+        final int n = (int)((end-start)/step);
+
+        if(n < 0) {
+            return new ${pt.primitive}[0];
+        }
+
+        final ${pt.primitive}[] result = new ${pt.primitive}[n+1];
+
+        for(int i=0; i<=n; i++){
+            result[i] = (${pt.primitive})(start + i*step);
+        }
+
+        return result;
+    }
+
+
+    <#if pt.valueType.isFloat >
+
+    /**
+     * Returns {@code true} if the value is NaN and {@code false} otherwise.
+     *
+     * @param value value.
+     * @return {@code true} if the value is NaN and {@code false} otherwise.
+     */
+    static public boolean isNaN(${pt.boxed} value) {
+        return isNaN(castDouble(value));
+    }
+
+    /**
+     * Returns {@code true} if the value is NaN and {@code false} otherwise.
+     *
+     * @param value value.
+     * @return {@code true} if the value is NaN and {@code false} otherwise.
+     */
+    static public boolean isNaN(${pt.primitive} value) {
+        return ${pt.boxed}.isNaN(value);
+    }
+
+    /**
+     * Returns {@code true} if the value is infinite and {@code false} otherwise.
+     *
+     * @param value value.
+     * @return {@code true} if the value is infinite and {@code false} otherwise.
+     */
+    static public boolean isInf(${pt.boxed} value) {
+        return isNaN(castDouble(value));
+    }
+
+    /**
+     * Returns {@code true} if the value is infinite and {@code false} otherwise.
+     *
+     * @param value value.
+     * @return {@code true} if the value is infinite and {@code false} otherwise.
+     */
+    static public boolean isInf(${pt.primitive} value) {
+        return ${pt.boxed}.isInfinite(value);
+    }
+
+    /**
+     * Returns {@code true} if the value is normal, where "normal" is defined as not infinite, not NaN, and not null.
+     *
+     * @param value value.
+     * @return {@code true} if the value is not infinite, NaN, nor null; {@code false} otherwise
+     */
+    static public boolean isNormal(${pt.boxed} value) {
+        return isNormal(castDouble(value));
+    }
+
+    /**
+     * Returns {@code true} if the value is normal, where "normal" is defined as not infinite, not NaN, and not null.
+     *
+     * @param value value.
+     * @return {@code true} if the value is not infinite, NaN, nor null; {@code false} otherwise
+     */
+    static public boolean isNormal(${pt.primitive} value) {
```

Comment:
If it's not too late I'd probably propose naming the method `isFinite` rather than `isNormal` (thus avoiding the problematic term Normal which means something else to floating-point people) and I'd probably propose writing
```
return ${pt.boxed}.isFinite(value) && !isNull(value)
```